torch/autograd/profiler.py

```diff
@@ -419,6 +449,11 @@ def _privateuse1_memory_usage(mem_record):
 
         # Create and return FunctionEvent list
         function_events = []
+        # prepare the no_link list to separate host events from with framework-assigned
+        # correlation IDs from other events whose correlation IDs were assigned
+        # concurrently but non-unique relative to the framework.
+        function_events_no_link = []
```

Comment:
@briancoutinho Would you please help review parts related to `function_events_no_link` which will same solve correlation IDs conflicts but without any search on list. It is set as a local list with limited lifecycle and has the same ability of `has_linked` but won't add a boring attribute into `FunctionEvent`.

Reply:
cc @aaronenyeshi for further review and re-merging. Sorry for the disturbance.